### PR TITLE
Exponential backoff + Create STS per bucket 

### PR DESCRIPTION
**Pull request by eshen1991**
1) Remove StsQuotaManager and rely on Google [ExponetialBackOff](https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/util/ExponentialBackOff) to retry when exceeding rating limit.

2) For dataset rule, create one STS job per bucket instead of per dataset rule. One bucket could have many dataset rules.

### CODE REVIEW

File: src/main/java/com/google/gcs/sdrs/controller/EventsController.java [hunk 1]
```diff
@@ -102,13 +103,27 @@ private void validateExecutionEvent(ExecutionEventRequest request) throws Valida
     } else {
       switch (request.getExecutionEventType()) {
         case USER_COMMANDED:
+          ValidationResult validateTargetNameResult =
+              FieldValidations.validateFieldFollowsBucketNamingStructure(
```
Comment (salguerod) [inline src/main/java/com/google/gcs/sdrs/controller/EventsController.java]: The case logic is getting a bit verbose, consider refactoring into its own helper method.

File: src/main/java/com/google/gcs/sdrs/service/rule/impl/StsRuleExecutor.java [hunk 2]
```diff
@@ -100,60 +101,92 @@ private StsRuleExecutor() throws IOException {
...
-      prefixes.add(prefix);
-    } else {
-      prefixes =
-          PrefixGeneratorUtility.generateTimePrefixes(
-              RetentionUtil.getDatasetPath(rule.getDataStorageName()),
-              zonedDateTimeNow.minusDays(lookBackInDays),
-              zonedDateTimeNow.minusDays(rule.getRetentionPeriodInDays()));
+      String sourceBucket = bucketName;
+      String destinationBucket = bucketName.concat(suffix);
+
```
Comment (salguerod) [inline src/main/java/com/google/gcs/sdrs/service/rule/impl/StsRuleExecutor.java]: white space

File: src/main/java/com/google/gcs/sdrs/service/worker/impl/ExecuteRetentionWorker.java [hunk 3]
```diff
@@ -62,6 +63,53 @@ public void doWork() {
...
+                  executionEvent.getExecutionEventType().toString()));
+      }
+      workerResult.setStatus(WorkerResult.WorkerResultStatus.SUCCESS);
+    } catch (IOException | IllegalArgumentException ex) {
+      logger.error(String.format("Error executing rule: %s", ex.getMessage()));
+      workerResult.setStatus(WorkerResult.WorkerResultStatus.FAILED);
+    }
+
+    /*   String dataStorageName = getDataStorageName(executionEvent.getTarget());
+    String projectId = executionEvent.getProjectId();
```
Comment (salguerod) [inline src/main/java/com/google/gcs/sdrs/service/worker/impl/ExecuteRetentionWorker.java]: Remove commented out code if not needed.

File: src/main/resources/applicationConfig.xml [hunk 4]
```diff
@@ -47,8 +47,6 @@
     <defaultProjectId>global-default</defaultProjectId>
     <defaultStorageName>global</defaultStorageName>
     <maxLookBackInDays>365</maxLookBackInDays>
-    <throttleLimit>30</throttleLimit>
```
Comment (salguerod) [inline src/main/resources/applicationConfig.xml]: can we update/remove this setting from the associated readme guide - https://github.com/GoogleCloudPlatform/storage-sdrs/blob/master/readme/README-executor.md

File: src/main/java/com/google/gcs/sdrs/controller/EventsController.java (hunk 1 above)
Comment (eshen1991) [reply src/main/java/com/google/gcs/sdrs/controller/EventsController.java]: done

File: src/main/java/com/google/gcs/sdrs/service/worker/impl/ExecuteRetentionWorker.java (hunk 3 above)
Comment (eshen1991) [reply src/main/java/com/google/gcs/sdrs/service/worker/impl/ExecuteRetentionWorker.java]: done

File: src/main/resources/applicationConfig.xml (hunk 4 above)
Comment (eshen1991) [reply src/main/resources/applicationConfig.xml]: done

File: src/main/java/com/google/gcs/sdrs/service/rule/impl/StsRuleExecutor.java (hunk 2 above)
Comment (eshen1991) [reply src/main/java/com/google/gcs/sdrs/service/rule/impl/StsRuleExecutor.java]: done

**Pull request merged**